File: app/src/main/java/com/nesib/countriesapp/adapters/CountryBordersAdapter.java
package com.nesib.countriesapp.adapters;

import android.net.Uri;
import android.os.Handler;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.github.twocoffeesoneteam.glidetovectoryou.GlideToVectorYou;
import com.nesib.countriesapp.R;
import com.nesib.countriesapp.models.Country;

import java.util.List;

public class CountryBordersAdapter extends RecyclerView.Adapter<CountryBordersAdapter.ViewHolder> {
    private List<Country> countryList;
    private OnItemClickListener listener;

    public void setListener(OnItemClickListener listener) {
        this.listener = listener;
    }

    public interface OnItemClickListener{
        void onClick(Country selectedCountry);
    }

    public CountryBordersAdapter(List<Country> countryList){
        this.countryList = countryList;
    }

    @NonNull
    @Override
    public CountryBordersAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.border_country_item,parent,false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull CountryBordersAdapter.ViewHolder holder, int position) {
        Country country = countryList.get(position);
        holder.countryName.setText(country.getName());
        GlideToVectorYou
                .init()
                .with(holder.itemView.getContext())
                .load(Uri.parse(country.getFlag()),holder.flagImage);
    }

    @Override
    public int getItemCount() {
        return countryList.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        public ImageView flagImage;
        public TextView countryName;
        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            flagImage = itemView.findViewById(R.id.borderCountryImage);
            countryName = itemView.findViewById(R.id.borderCountryName);
            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    Animation scaleOut = AnimationUtils.loadAnimation(view.getContext(),R.anim.scale_out);
                    itemView.startAnimation(scaleOut);
                    new Handler().postDelayed(new Runnable() {
                        @Override
                        public void run() {
                            listener.onClick(countryList.get(getAdapterPosition()));
                        }
                    },200);

                }
            });
        }
    }


}

File: app/src/main/java/com/nesib/countriesapp/adapters/CountryBordersAdapter.kt
package com.nesib.countriesapp.adapters

import android.net.Uri
import android.os.Handler
import com.nesib.countriesapp.models.Country
import androidx.recyclerview.widget.RecyclerView
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import com.nesib.countriesapp.R
import com.github.twocoffeesoneteam.glidetovectoryou.GlideToVectorYou
import android.widget.TextView
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import android.widget.ImageView

class CountryBordersAdapter(private val countryList: List<Country>) :
    RecyclerView.Adapter<CountryBordersAdapter.ViewHolder>() {
    private var listener: OnItemClickListener? = null
    fun setListener(listener: OnItemClickListener?) {
        this.listener = listener
    }

    interface OnItemClickListener {
        fun onClick(selectedCountry: Country)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view =
            LayoutInflater.from(parent.context).inflate(R.layout.border_country_item, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val country = countryList[position]
        holder.countryName.text = country.name
        GlideToVectorYou
            .init()
            .with(holder.itemView.context)
            .load(Uri.parse(country.flag), holder.flagImage)
    }

    override fun getItemCount(): Int {
        return countryList.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var flagImage: ImageView = itemView.findViewById(R.id.borderCountryImage)
        var countryName: TextView = itemView.findViewById(R.id.borderCountryName)

        init {
            itemView.setOnClickListener { view ->
                val scaleOut = AnimationUtils.loadAnimation(view.context, R.anim.scale_out)
                itemView.startAnimation(scaleOut)
                Handler().postDelayed(Runnable {
                    listener!!.onClick(
                        countryList[adapterPosition]
                    )
                }, 200)
            }
        }
    }
}